app/experimenter/experiments/models/nimbus.py

```diff
@@ -426,12 +427,29 @@ def latest_review_request(self):
         return (
             self.all()
             .filter(
+                old_status=NimbusExperiment.Status.DRAFT,
                 old_publish_status=NimbusExperiment.PublishStatus.IDLE,
+                new_status=NimbusExperiment.Status.DRAFT,
                 new_publish_status=NimbusExperiment.PublishStatus.REVIEW,
             )
             .order_by("-changed_on")
         ).first()
 
+    def latest_rejection(self):
```

Comment:
Don't we want this to be per-experiment, rather than the latest rejection across the system? Otherwise, then won't every draft experiment show a rejection if one gets rejected for review?

Reply:
When you call it from an experiments related manager it applies the per experiment filter

`experiment.changes.latest_rejection()` expands to

```sql
 SELECT "experiments_nimbuschangelog"."id",
       "experiments_nimbuschangelog"."experiment_id",
       "experiments_nimbuschangelog"."changed_on",
       "experiments_nimbuschangelog"."changed_by_id",
       "experiments_nimbuschangelog"."old_status",
       "experiments_nimbuschangelog"."old_publish_status",
       "experiments_nimbuschangelog"."new_status",
       "experiments_nimbuschangelog"."new_publish_status",
       "experiments_nimbuschangelog"."message",
       "experiments_nimbuschangelog"."experiment_data"
FROM   "experiments_nimbuschangelog"
WHERE  ( "experiments_nimbuschangelog"."experiment_id" = 8
         AND ( "experiments_nimbuschangelog"."old_publish_status" = 'Review'
                OR "experiments_nimbuschangelog"."old_publish_status" =
                   'Waiting' )
         AND "experiments_nimbuschangelog"."new_publish_status" = 'Idle'
         AND "experiments_nimbuschangelog"."new_status" = 'Draft'
         AND "experiments_nimbuschangelog"."old_status" = 'Draft' )
ORDER  BY "experiments_nimbuschangelog"."changed_on" DESC
LIMIT  1;  
```

Reply:
Ah, I forgot about that magic bit of Django ORM